### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by radeklos**
This is simple PR which has simple core of the app.. It has some http handlers, routers and serialisers and most of the thinks are coverage by tests. If there is something which is not really `go` let me know :)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM alpine:3.2
+
+RUN echo 'http://dl-4.alpinelinux.org/alpine/edge/main' >> /etc/apk/repositories \
```
Comment (krak3n) [inline Dockerfile:11]: Only need to do this if we wanna run this in go 1.5 - 1.4 should be fine and thats whats in the stable packages.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    && apk update && apk add go git ca-certificates make && rm -rf /var/cache/apk/* \
+    && wget -P /usr/local/bin http://public.thisissoon.com.s3.amazonaws.com/glide \
+    && chmod +x /usr/local/bin/glide
+
+ENV GOPATH=/deadpool
+
+RUN go get github.com/gorilla/mux
```
Comment (krak3n) [inline Dockerfile]: I'd certainly look at using `gpm` as we do on Myleene and other go projects to manage dependencies during the build.

**Comment by radeklos** [comment]
@krak3n @peeklondon CR please.

**Comment by krak3n** [comment]
I'll check it out in the :sunrise:

File: Godeps [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#
+# Project Dependencies
+#
+
+github.com/gorilla/mux              master
```
Comment (krak3n) [inline Godeps]: You can also pass in commit refs - if you use master you could get breaking changes form build to build.

File: serializer/serializer.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+type Report struct {
+	HttpRequest  string `json:"request"`
+	HttpResponse string `json:"respond"`
+	Source       string `json:"source"`
+	Destination  string `json:"destination"`
+	Description  string `json:"description"`
+}
+
+func UnmarshalReport(data []byte, r *Report) error {
+	defer func() {
```
Comment (krak3n) [inline serializer/serializer.go]: A different way might be to do something like this:

``` go
// Acts as a constructor for a new report
func NewReport(data []byte) (*Report, error) {
    r := &Report{}
    // put your defered panic catcher here
    if err := json.Unmarshal(data, r); err != nil {
        return nil, err
    }

    return r, nil
}
```

Then other parts of the apps can just call `r, err = serializer.NewReport(data)`

You could then hang of a `serialize` function off the `Report` type that serialized the report to json:

``` go
func (r *Report) ToJSON()([]byte, err) {
     return json.Marshal(r)
}
```

**Comment by krak3n** [comment]
Great work!

I suggested an alternative way of constructing new reports but the way you have done it is fine anyway. 

:+1:

**Pull request merged**